src/inventory_ui.cpp

```diff
@@ -2623,9 +2640,43 @@ void inventory_drop_selector::set_chosen_count( inventory_entry &entry, size_t c
 
 inventory_selector::stats inventory_drop_selector::get_raw_stats() const
 {
-    return get_weight_and_volume_stats(
-               u.weight_carried_with_tweaks( dropping ),
-               u.weight_capacity(),
-               u.volume_carried_with_tweaks( dropping ),
-               u.volume_capacity_with_tweaks( dropping ) );
+    stats drop_stats = get_weight_and_volume_stats(
+                           u.weight_carried_with_tweaks( dropping ),
+                           u.weight_capacity(),
+                           u.volume_carried_with_tweaks( dropping ),
+                           u.volume_capacity_with_tweaks( dropping ) );
+
+    std::pair<units::mass, units::volume> drop_weight_volume = get_drop_weight_volume();
+
+    stat &weight_stats = drop_stats.front();
+    weight_stats.push_back( " -> " );
+    weight_stats.push_back( colorize( string_format( "%.1f",
+                                      round_up( convert_weight( drop_weight_volume.first ), 1 ) ), c_light_gray ) );
+
+    stat &volume_stats = drop_stats.back();
+    volume_stats.push_back( " -> " );
+    volume_stats.push_back( colorize( format_volume( drop_weight_volume.second ), c_light_gray ) );
+
+    return drop_stats;
+}
+
+std::vector<alignment> inventory_drop_selector::get_stats_alignment() const
+{
+    // <Caption>," ",<Units>,"/",<Units>," -> ",<Units>
+    return {alignment::RIGHT, alignment::LEFT, alignment::RIGHT, alignment::LEFT, alignment::LEFT, alignment::LEFT, alignment::RIGHT};
+}
+
+std::pair<units::mass, units::volume> inventory_drop_selector::get_drop_weight_volume() const
+{
+    units::mass drop_mass = 0_gram;
+    units::volume drop_volume = 0_ml;
+    for( const std::pair<item_location, int> &loc_pair : dropping ) {
+        const item *dropped_item = loc_pair.first.get_item();
+        if( dropped_item ) {
+            drop_mass += dropped_item->weight() * loc_pair.second;
+            drop_volume += dropped_item->volume() * loc_pair.second;
```

Comment:
`item::volume` already accounts for charges, so for item stacks you multiply by charges twice here.

Reply:
Got it, I've tested stacks of simple items, loaded/unloaded magazines and single separate items, seems it works correctly now.